modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v4_0_3/CommerceRepositoryUpgradeProcess.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.internal.upgrade.v4_0_3;
+
+import com.liferay.commerce.product.constants.CPConstants;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.RepositoryLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.Portal;
+
+/**
+ * @author Michael Bowerman
+ */
+public class CommerceRepositoryUpgradeProcess extends UpgradeProcess {
+
+	public CommerceRepositoryUpgradeProcess(
+		CompanyLocalService companyLocalService, Portal portal,
+		RepositoryLocalService repositoryLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_portal = portal;
+		_repositoryLocalService = repositoryLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasTable("ChangesetEntry")) {
+			return;
+		}
+
+		long classNameId = _portal.getClassNameId(Repository.class.getName());
+
+		_companyLocalService.forEachCompanyId(
+			companyId -> _upgradeCommerceRepository(companyId, classNameId));
+	}
+
+	private void _upgradeCommerceRepository(long companyId, long classNameId)
+		throws Exception {
+
+		Company company = _companyLocalService.getCompany(companyId);
+
+		Repository repository = _repositoryLocalService.fetchRepository(
+			company.getGroupId(), "image.default.company.logo",
+			CPConstants.SERVICE_NAME_PRODUCT);
+
+		if (repository == null) {
+			return;
```

Comment:
Shouldn't this be a continue clause?

Reply:
Hi @andrea-ale-sbarra, sorry, I'm afraid I don't really understand what you're asking. As far as I understood, I thought the `continue` keyword in Java was only used for `for` loops or `while` loops.

Reply:
yes you're right indeed, and this function is run in a forEach method. I'm not sure with the return here it would continue to the next company. So basically did you try this with multiple company and the first of these companies returning a null repository?

Reply:
```
	public static void main(String[] args) {
        List<String> strings = Arrays.asList(new String[]{"uno", "due", "tre"});

        strings.forEach(s -> {
            if(s.equals("uno")){
                return;
            }
            System.out.println(s);
        });
    }
}
```

Reply:
I've made this simple test, and it is printing the due and tre strings. So I had a doubt but the code is working as expected.
Thank you!